apps/server/src/shared/domain/rules/course-group.rule.spec.ts

```diff
@@ -2,8 +2,8 @@ import { Test, TestingModule } from '@nestjs/testing';
 import { CourseGroup, User } from '@shared/domain/entity';
 import { Permission } from '@shared/domain/interface';
 import { courseFactory, courseGroupFactory, roleFactory, setupEntities, userFactory } from '@shared/testing';
-import { AuthorizationHelper } from '../authorization.helper';
-import { Action } from '../types';
+import { AuthorizationHelper } from '../../../modules/authorization/authorization.helper';
```

Comment:
Can you please export authorization.helper and types over the index file of modules/authorization/index.ts and import it from this? Maybe with @src/modul.

Also for the next imports in the other spec files.

Reply:
Das hat gerade zu interessanten Problemen geführt. In den Test-Files kann man das tatsächlich so machen, aber in den eigentlichen Rules nicht. Dann kann Nest die Dependency nicht mehr auflösen, weil die Rules Teil des Authorization-Moduls sind und man im Modul selbst nicht das Modul importieren darf, siehe: https://github.com/nestjs/nest/issues/9095

Da war ich gerade erstmal ziemlich verwirrt, weil das kaschiert wird dadurch, dass die Rules in shared liegen. Ich finde, das ist nochmal ein Argument, sie auch in der Ordner-Struktur ins Authz-Modul zu packen.

Und der Helper soll eigentlich auch nur Modul-intern benutzt und nicht exportiert werden.

Reply:
Die dependency von Authorisation Modul zeigt auf jedes fachliche Modul in de dieser verwendet wird. Daher dürfen Module die das Modul importieren auch die exports von diesem nutzen. Da die Rules irgendwann dort auch liegen, dürfen sie den Helper verwenden. 
Sie dürfen natürlich nicht wie jetzt direkt dort hinzugefügt werden, daher muss die Dependency auch noch umgedreht werden.

Es ist wie vor dem PR nach wie vor nicht sauber. 

Wenn er jetzt erstmal die Shared nicht mag ist das ok. Wobei mir nicht ganz klar ist wo er da meckert. Ich hatte vermutet das evtl. ein nodejs dependency cicle irgendwo auftreten könnte. Aber für nestjs dürfte der relative Pfad vs dem absoluten kein unterschied machen um zu identifizieren was zu ein Modul gehört.

Reply:
In dem zukünftigen Stand mag das Sinn ergeben, den Helper zu exportieren, momentan tut es das nicht, weil die Rules Teil des Authorization-Modules sind und der Helper nur von den Rules verwendet wird. Ich finde es nicht sinnvoll, immer von einem (möglichen) zukünftigen Stand aus zu argumentieren.

Das Problem mit den Rules in shared ist, dass es für den Leser verdeckt, dass die Rules eigentlich Teil des Authorization-Modules sind. Dadurch hab ich dieses Problem mit den Imports nicht gleich verstanden. Das Problem war, um es nochmal anders zu formulieren: Innerhalb eines Moduls muss man direkt aus den einzelnen Files importieren (so sagt es zumindest der verlinkte Github-Thread und unser Fehler hier). Da die Rules Teil des Authorization-Modules sind, muss man den Helper direkt aus dem File importieren, aus "@src/modules/authorization/authorization-helper" und kann ihn nicht aus "@src/modules/authorization" importieren.